configure.ac

```diff
@@ -1351,6 +1351,36 @@ AC_SUBST(OPENTHREAD_ENABLE_SERVICE)
 AM_CONDITIONAL([OPENTHREAD_ENABLE_SERVICE], [test "${enable_service}" = "yes"])
 AC_DEFINE_UNQUOTED([OPENTHREAD_ENABLE_SERVICE],[${OPENTHREAD_ENABLE_SERVICE}],[Define to 1 if you want to enable Service])
 
+#
+# Time Synchronization Service
+#
+
+AC_ARG_ENABLE(time_sync,
+    [AS_HELP_STRING([--enable-time-sync],[Enable Time Synchronization support @<:@default=no@:>@.])],
```

Comment:
Is it necessary to expose this proprietary feature as a configure option?

Reply:
I think it's not really necessary, I have updated the PR to only introduce a new macro definition: OPENTHREAD_CONFIG_HEADER_IE_SUPPORT, it will be used to enable/disable Header IE generation and parsing feature separately.

Reply:
@jwhui How to determine if a feature should provide configure option? It seems some proprietary features also provides configure options, such as raw link, channel monitor, and so on.

Reply:
In general, configure options tend to be more visible to developers and offer less opportunity for inline documentation.  For that reason, I believe features that can cause incompatibility with standard Thread 1.1 implementations should not be configure options.  The current set of configure options, including the ones you listed, do not cause incompatibility with standard Thread 1.1 implementations.  Thoughts?

Reply:
Got it. I used to think configure options provide options to include optional features or not while inline configurations provide tweaks about how features behave. Thanks for your clarification.